packages/codegen/src/generateMetadataFile.ts

```diff
@@ -46,15 +47,27 @@ function generateColumns(dbMetadata: EntityDbMetadata): {
   `;
 
   const primitives = dbMetadata.primitives.map((p) => {
-    const { fieldName, columnName, columnType } = p;
+    const { fieldName, columnName, columnType, superstruct } = p;
     const serdeType = columnType === "numeric" ? DecimalToNumberSerde : SimpleSerde;
-    return code`
-      {
-        fieldName: "${fieldName}",
-        columnName: "${columnName}",
-        dbType: "${columnType}",
-        serde: new ${serdeType}("${fieldName}", "${columnName}"),
-      },`;
+    if (superstruct) {
```

Comment:
If a primitive field is a jsonb/backed by a superstruct, we go through a different serde helper that will apply the superstruct `assert` function to any value from the db, to make sure it matches what we expect.